### DISCUSSION TITLE
How to implement local gradient in a tensor

### DISCUSSION THREAD

**Discussion by xiaolang01** [Lightning-AI/pytorch-lightning#13428] [code help: NLP / ASR / TTS]
![image](https://user-images.githubusercontent.com/44439592/176147819-3798c6d7-4127-4061-84fd-a6b0d3be7274.png)
I want to achieve such an operation how the code should be written？

**Comment by akihironitta** [comment]
You should be able to achieve it by overriding `on_after_backward` with the logic. Something like this:

```python
def on_after_backward(self):
    for p in self.model...parameters():
        p.grad[:original_vocab_size, :] = 0.0
```

https://pytorch-lightning.readthedocs.io/en/1.6.4/common/lightning_module.html#on-after-backward
